src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -1231,7 +1231,7 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
     }
 
     assert(finalNames.areDistinct())
-    val touchesKeys = rowKey.forall(topLevelFields.contains)
+    val touchesKeys = !rowKey.forall(topLevelFields.contains)
```

Comment:
not `forall` is exists

Reply:
Actually this is like intersection of sets, which might read more straightforwardly.